application/models/GoogleSheets_Model.php

```diff
@@ -108,6 +108,60 @@ public function addNewRecord($email, $fullName, $paymentMethod)
         $result = $this->service->spreadsheets_values->update($this->spreadsheetId, $range, $body, $params);
     }
 
+    /**
+     * Mark an attendees attendance in the google sheets
+     *
+     * @param string $eventName The name of the event
+     * @param string $email The email of the user to record.
+     * @param string @upi The upi of the user if it exists
+     */
+    public function markAsPresent($eventName, $email = null, $upi = null)
+    {
+        // This should probably be in a try/catch block
+        if (is_null($email) and if_null($upi)) {
```

Comment:
Instead of checking null, we can check `if (!$email && !$upi ||) ...` instead. This way, we ensure that one of `$email` or `$upi` must be true (i.e. exist and is a string with characters).

The reason why this is useful is because `null` could not be the only value passed, i.e. we could potentially get a string of 0 characters.

See this stackoverflow for other falsey value equivalents: https://stackoverflow.com/questions/2382490/how-does-true-false-work-in-php